dali/operators/random/noise/salt_and_pepper_noise.h

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_RANDOM_NOISE_SALT_AND_PEPPER_NOISE_H_
+#define DALI_OPERATORS_RANDOM_NOISE_SALT_AND_PEPPER_NOISE_H_
+
+#include "dali/operators/random/rng_base.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/operators/random/rng_base_gpu.h"
+#include "dali/operators/random/rng_base_cpu.h"
+
+#define DALI_SALT_AND_PEPPER_NOISE_TYPES \
+  (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float)
+
+namespace dali {
+
+template <typename Backend, typename T>
+class salt_and_pepper_noise_impl {
+ public:
+  using Dist = typename std::conditional_t<std::is_same<Backend, GPUBackend>::value,
+                                           curand_uniform_dist<float>,
+                                           std::uniform_real_distribution<float>>;
+
+  DALI_HOST_DEV explicit salt_and_pepper_noise_impl(float noise_prob = 0.05,
+                                                    float salt_to_pepper_prob = 0.5)
+      : noise_prob_(clamp<float>(noise_prob, 0, 1)),
+        salt_prob_(noise_prob_ * clamp<float>(salt_to_pepper_prob, 0, 1)),
+        salt_val_(ConvertNorm<T>(1.0f)),
+        pepper_val_(std::is_unsigned<T>::value ? ConvertSatNorm<T>(0.0f) :
+                                                 ConvertSatNorm<T>(-1.0f)) {}
```

Comment:
So the assumption is that float images has range from -1 to 1, what if someone has uint8 based image and then convert it to float?